platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,387 @@
+                       @[@(12*MGLFeetPerMile), @2],
+                       @[@(15*MGLFeetPerMile), @3],
+                       @[@(20*MGLFeetPerMile), @2],
+                       @[@(30*MGLFeetPerMile), @3],
+                       @[@(40*MGLFeetPerMile), @2],
+                       @[@(80*MGLFeetPerMile), @2],
+                       @[@(120*MGLFeetPerMile), @2],
+                       @[@(200*MGLFeetPerMile), @2],
+                       @[@(300*MGLFeetPerMile), @3],
+                       @[@(400*MGLFeetPerMile), @2],
+                       @[@(800*MGLFeetPerMile), @2],
+                       @[@(1000*MGLFeetPerMile), @2],
+                       @[@(2000*MGLFeetPerMile), @2],
+                       @[@(3000*MGLFeetPerMile), @3],
+                       @[@(4000*MGLFeetPerMile), @2]];
+}
+
+#pragma mark - Dimensions
+
+- (CGSize)intrinsicContentSize {
+    return CGSizeMake(self.actualWidth, 16);
+}
+
+- (CGFloat)preferredWidth {
+    CGFloat padding = CGRectGetMinX(self.frame);
+    return ([self maximumWidth] - padding) / 2;
+}
+
+- (CGFloat)actualWidth {
+    return floorf(self.row.firstObject.floatValue / [self unitsPerPoint]);
+}
+
+- (CGFloat)minimumWidth {
+    CGFloat fullWidth = CGRectGetWidth(self.superview.bounds);
+    CGFloat padding = CGRectGetMinX(self.frame);
+    CGFloat width = fullWidth - padding;
+    return floorf((width / 2.0) / 3.0);
+}
+
+- (CGFloat)maximumWidth {
+    CGFloat fullWidth = CGRectGetWidth(self.superview.bounds);
+    CGFloat padding = CGRectGetMinX(self.frame);
+    return floorf(fullWidth / 2 - padding);
+}
+
+- (CGFloat)unitsPerPoint {
+    return [self usesMetricSystem] ? self.metersPerPoint : self.metersPerPoint * MGLFeetPerMeter;
+}
+
+#pragma mark - Convenient methods
+
+- (NSArray<NSArray<NSNumber *> *> *)validRows {
+    CLLocationDistance minimumDistance = [self minimumWidth] * [self unitsPerPoint];
+    CLLocationDistance maximumDistance = [self maximumWidth] * [self unitsPerPoint];
+    
+    NSMutableArray *validRows = [NSMutableArray array];
+    NSArray *table = [self usesMetricSystem] ? self.metricTable : self.imperialTable;
+    for (NSArray<NSNumber *> *row in table) {
+        CLLocationDistance distance = row.firstObject.floatValue;
+        if (distance >= minimumDistance && distance <= maximumDistance) {
+            [validRows addObject:row];
+        }
+    }
+    
+    return validRows;
+}
+
+// Find the row closest to -[MGLScaleBarView preferredWidth]
+- (NSArray<NSNumber *> *)preferredRowInRows:(NSArray<NSArray<NSNumber *> *> *)rows {
+    CLLocationDistance diff = CGFLOAT_MAX;
+    CLLocationDistance preferredDistance = [self preferredWidth] * [self unitsPerPoint];
+    NSArray<NSNumber *> *preferredRow = rows.firstObject;
+    for (NSArray<NSNumber *> *row in rows) {
+        CLLocationDistance currentDiff = ABS(preferredDistance - row.firstObject.floatValue);
+        if (currentDiff < diff) {
+            preferredRow = row;
```

Comment:
Update diff is missing here so it always prefers the last row which looks better and doesn't make the labels overlap. This method can be removed then always pick the last row from -validRows.